rustls/examples/internal/bench_impl.rs

```diff
@@ -491,14 +498,24 @@ fn bench_bulk(
     let rounds = total_data / plaintext_size;
 
     if options.api.use_buffered() {
+        let threads = (0..options.threads.into())
+            .map(|_| {
+                let client_config = client_config.clone();
+                let server_config = server_config.clone();
+                thread::spawn(move || {
+                    bench_bulk_buffered(client_config, server_config, plaintext_size, rounds)
```

Comment:
Worth extracting this as a function parametrized with an `impl Fn(..)`?

(Might even consider making it a trait...)